### ISSUE TITLE
Issue with Serial number

### ISSUE THREAD

**Issue by irtimmer** [alonbl/gnupg-pkcs11-scd#6]
Trying to generate a key based on an existing key results in a repeated message from pinentry (tested with GnuPG 2.1.19 (and 2.1.20) and gnupg-pkcs11-scd 0.7.5):
```
Please remove the current card and insert the one with serial number:
D2760001240111504B4353233131111100
```

In the gpg-agent.log I could only find gpg-agent demanding for this card-number and gnupg-pkcs11-scd responding with a similar looking serial number:
```
2017-04-15 17:58:58 gpg-agent[10393] no running SCdaemon - starting it
2017-04-15 17:58:58 gpg-agent[10393] DBG: chan_10 <- OK PKCS#11 smart-card server for GnuPG ready
2017-04-15 17:58:58 gpg-agent[10393] DBG: first connection to SCdaemon established
2017-04-15 17:58:58 gpg-agent[10393] DBG: chan_10 -> GETINFO socket_name
2017-04-15 17:58:58 gpg-agent[10393] DBG: chan_10 <- D /tmp/gnupg-pkcs11-scd.6S6lKI/agent.S
2017-04-15 17:58:58 gpg-agent[10393] DBG: chan_10 <- OK
2017-04-15 17:58:58 gpg-agent[10393] DBG: additional connections at '/tmp/gnupg-pkcs11-scd.6S6lKI/agent.S'
2017-04-15 17:58:58 gpg-agent[10393] DBG: chan_10 -> OPTION event-signal=12
2017-04-15 17:58:58 gpg-agent[10393] DBG: chan_10 <- OK
2017-04-15 17:58:58 gpg-agent[10393] DBG: chan_10 -> SERIALNO --demand=D2760001240111504B4353233131111100
2017-04-15 17:58:58 gpg-agent[10393] DBG: chan_10 <- S SERIALNO D2760001240111504B43532331311111 0
2017-04-15 17:58:58 gpg-agent[10393] DBG: chan_10 <- OK
2017-04-15 17:58:58 gpg-agent[10393] DBG: detected card with S/N D2760001240111504B43532331311111
2017-04-15 17:58:58 gpg-agent[10393] starting a new PIN Entry
2017-04-15 17:58:58 gpg-agent[10393] DBG: connection to PIN entry established
2017-04-15 17:58:58 gpg-agent[10393] DBG: chan_9 -> INQUIRE PINENTRY_LAUNCHED 10400 qt 1.0.0 ? ? ?
2017-04-15 17:58:58 gpg-agent[10393] DBG: chan_9 <- END
2017-04-15 17:59:00 gpg-agent[10393] DBG: chan_10 -> SERIALNO --demand=D2760001240111504B4353233131111100
2017-04-15 17:59:00 gpg-agent[10393] DBG: chan_10 <- S SERIALNO D2760001240111504B43532331311111 0
2017-04-15 17:59:00 gpg-agent[10393] DBG: chan_10 <- OK
2017-04-15 17:59:00 gpg-agent[10393] DBG: detected card with S/N D2760001240111504B43532331311111
2017-04-15 17:59:00 gpg-agent[10393] starting a new PIN Entry
2017-04-15 17:59:00 gpg-agent[10393] DBG: connection to PIN entry established
2017-04-15 17:59:00 gpg-agent[10393] DBG: chan_9 -> INQUIRE PINENTRY_LAUNCHED 10403 qt 1.0.0 ? ? ?
2017-04-15 17:59:00 gpg-agent[10393] DBG: chan_9 <- END
2017-04-15 17:59:02 gpg-agent[10393] DBG: error calling pinentry: Bewerking geannuleerd <Pinentry>
2017-04-15 17:59:02 gpg-agent[10393] smartcard signing failed: Bewerking geannuleerd
2017-04-15 17:59:02 gpg-agent[10393] command 'PKSIGN' failed: Bewerking geannuleerd <Pinentry>
2017-04-15 17:59:02 gpg-agent[10393] DBG: chan_9 -> ERR 83886179 Bewerking geannuleerd <Pinentry>
2017-04-15 17:59:02 gpg-agent[10393] DBG: chan_9 <- [eof]
2017-04-15 17:59:02 gpg-agent[10393] DBG: chan_10 -> RESTART
2017-04-15 17:59:02 gpg-agent[10393] DBG: chan_10 <- OK
```
At the changelog is stated there were some changes in handling of the serial number for GnuPG 2.1.19. But it still doesn't seem to work on my setup.

**Comment by alonbl** [comment]
Hi,
Once again a change in the gnupg interface since 2.1.19 pre-releast to 2.1.19 or 2.1.20.
It adds 00 to the serial number.
Will need to manage this as well.
Thanks!

**Comment by alonbl** [comment]
Root cause[1]

[1] https://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=commit;h=79cea89774e6327b6785e22b7057f9e3e188ac2b

**Comment by alonbl** [comment]
Can you please check master?
You should be able to remove card then gpg --card-status then add card then gpg --card-status.
If persists, kill gpg-agent and remove ~/.gnupg/private-keys-v1.d/<hexgrip>.key of your private keys, you can obtain these using gpg-agent --server connect-agent and issue SCD LEARN.

**Comment by irtimmer** [comment]
Thanks, after updating and removing the key file it worked

**Comment by alonbl** [comment]
Thanks, I released 0.7.6.
